eslint-rules/eslint-process-env.mjs

```diff
@@ -0,0 +1,61 @@
+export default {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'Disallow usage of process.env outside config.js'
+    },
+    schema: []
+  },
+  create (context) {
+    // eslint-disable-next-line @stylistic/js/max-len
+    const allowedFile = /[/\\]packages[/\\]dd-trace[/\\]src[/\\](config-helper|guardrails[/\\](index|log|telemetry))\.js$/
```

Comment:
Why not just add `/* eslint-disable eslint-rules/eslint-process-env */` at the top of these files?

Reply:
No strong opinion. I believe we should not lint the guardrails in general. We can also overcome that differently though.